### DISCUSSION TITLE
Support for Yahoo and Other Leagues

### DISCUSSION THREAD

**Discussion by nmelhado** [nmelhado/league-page#145] [Ideas]
How to broaden the scope of League Page to support yahoo leagues as well.

**Comment by nmelhado** [comment]
@cuddebtj moving this to its own discussion

**Comment by nmelhado** [comment]
To do this, we would have to make wrapper functions to translate the responses from yahoo to make them the same shape as the Sleeper responses. I haven't dug into the Yahoo API at all, but I think this would be the biggest hurtle.

Beyond that, we would just add an enum option for users to select their provider and call the corresponding call for that provider.

I don't have the bandwidth to do and test the first part. But if someone else can tackle that part, I can jump in and finish the second part. I might even be able to do the second part first and leave TODOs for where the rest of the work needs to be done.

**Reply by cuddebtj** [reply]
There is a really well documented yahoo wrapper by uberfastman call yfpy. I've been creating a wrapper around that to extract that data into nice pandas df's (still a work in progress). I'll look at the data structure from sleeper and format it similarly, Yahoo does require authorization/a registered app for someone to access their league. An additional step in setup for Yahoo leagues would be needed.

**Reply by nmelhado** [reply]
If people are willing to do the extra work, that's fine. The blog instructions also require people to create keys and save the secret, so as long as the steps are well documented I think it'll be fine.

The only trick will be doing this in js. I'm sure there's a js/node wrapper as well, but I haven't looked into it. That's the part I won't have time to play around with.

I will try to prioritize making this provider agnostic and leaving some todos for other providers (yahoo and espn come to mind). That way you (and maybe others) can start working on the rest of the integrations.

**Reply by jackb22** [reply]
Have you made any progress on this? Willing to help out where you need it.
